client/src/main/java/io/pravega/client/stream/ReaderGroup.java

```diff
@@ -177,6 +177,15 @@
      */
     ReaderSegmentDistribution getReaderSegmentDistribution();
 
+    /**
+     * The {@link StreamCut} provided to this API indicates that the stream has consumed data upto
```

Comment:
`indicates that the stream has consumed data`
This is implying a semantic about how the user's application works. We don't know anything about their application. Instead we should define the API in terms of what the effect is.

Reply:
> stream has consumed data upto

secondly, its not the stream that consumes data but the readergroup that consumes from the stream. 
you should effectively say something to the effect of - the stream may get truncated at the streamcut passed to this api and only data beyond this streamcut may be retained.

Reply:
I've rephrased this in terms of retention. Would saying "stream has processed data" make more sense?